### ISSUE TITLE
Mvc samples won't run - all routes returning 404

### ISSUE THREAD

**Issue by ciel** [aspnet/Mvc#4321]
Trying the latest bits, I've downloaded the repository - updated the dot net cli to the latest version I can find (1.0.0-rc2-20221) and run **dotnet restore**, **dotnet build**, and **dotnet run** on the sample; It compiles fine, it launches fine, but a visit to **http://localhost:5000** just yields a 404 error in the debugging tools in the browser.

This is happening to other web applications I've got too, that I've fixed and updated to reflect the newest pieces. It happens even when I set the **UseContentRoot** method on the **WebHostBuilder** correctly.

Is this known? Or is there a new way of compiling/launching these suddenly? I'm presently trying to get the normal **MvcSample** to run, but this happens with all of them.

**Comment by ciel** [comment]
I got this solved through two solid days of extremely frustrating trial and error. I'm sharing my story in hopes that others will not experience the same problem.

First of all, I've tried to be diligent about keeping the `dot net cli` up to date. Every morning, I `git clone` the `dotnet cli` repository, and run `cli/scripts/obtain/install.ps1 -channel beta` to get the newest version of the `cli`. Then, I run `dnvm upgrade -u latest -r clr -arch x64` to make sure I've got the newest build of the whole thing.

Then I clear out all of my caches and package stores using the following `.bat` file;
```
start rimraf %localappdata%\NuGet\v3-cache
start rimraf %userprofile%\.nuget\packages
start rimraf %userprofile%\.dnx\packages
```

Then I clear out all of the `/bin`, `/obj`, and `/artifacts` folders in my solutions using the following `.bat` file;
```
start rimraf Projects/**/src/**/bin
start rimraf Projects/**/src/**/obj
start rimraf Projects/**/src/**/artifacts/*
```
This gets everything cleared out, all gunk removed, etc.

Then I go to the root of the project I'm working on and run `dotnet restore`, and my project(s) have the following `nuget.config`;
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <add key="aspnet-core" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
        <add key="nuget" value="https://api.nuget.org/v3/index.json" />
    </packageSources>
</configuration>
```
So at this point, I'm ready to build and run! After the (eternal) wait of `dotnet restore`, I go to the `Web` project and run `dotnet build`, `dotnet run`, and get things loaded and running! 

This worked well until this week, when I was met with the problem of `http://localhost:5000/` returning `404` on everything I tried. I spent a long time working at it, and just couldn't find the problem until I made a _mistake_.

First, I decided to try the latest `.msi` build of the dot net `cli`, instead of obtaining it through the `install.ps1 -channel beta` command. The difference in the build numbers was immediately apparent, and the `.msi` file was actually a lower version - so on a hunch, I went with the lower version instead of the 'edge' release. 

At the time of this writing, the version of the `.NET Command Line Tools` was `1.0.0-beta-001897`. The version of the `.NET Version Manager` was `v1.0.0-rc2-15546`, and the version of the runtime I had set to `default` was `1.0.0-rc2-20221 clr x64` on `Windows 10 Professional x64`.

This time, after updating the `dotnet cli` to the... older version, I tried to run it and was given a warning by my antivirus - which I suspect was a culprit in the entire problem the whole time. It was `Ransomware Protection`, and was not allowing the `dotnet` application to write to the folders. I looked back at my system history, and tried updating to the `-channel beta` version again just to test this out, and sure enough it was happening there too - but I was never being prompted for it.

After disabling that, and properly adding all of the `dotnet` folder `.exe` files to the "whitelist" on my antivirus, I ran the entire setup again using the latest `.msi` version of the `cli` (again, `1.0.0-beta-001897`) and it started to work correctly!

Allowing the various `.exe` files for the `-channel beta` build of the command line tools continued to yield the `404` errors from Kestrel. Even turning off the antivirus completely did not fix that problem.

So in all, I traced down two issues at play here;

1. the `dot net cli` from the `beta` channel was having trouble
2. antivirus blocked the `dotnet.exe` application and others in its folder for some reason

I'm uncertain how much of this was due to the antivirus, I think the big problem is that I never got an alert for it until I ran the `.msi` installer for the `dotnet cli`. I've tried to reproduce this whole scenario on a `Hyper-V` virtual machine using `Windows 8.1` and `Windows 10` and was easily able to both times by using the `beta` channel of the `cli`.

My only guess is that there is something wrong with the edge build of the cli, or the kestrel server isn't yet updated to compile against something it checks (the new framework names, perhaps? I'm uncertain)

I don't think this is inherently a problem with the `MVC` libraries, but I'm leaving the comment here since this was the first place I came when I began encountering the trouble. I hope someone else can learn from it, or at least tell me what I did wrong.
